addons/project/models/project_task.py

```diff
@@ -1385,6 +1383,33 @@ def _task_message_auto_subscribe_notify(self, users_per_task):
                     model_description=task_model_description,
                     mail_auto_delete=False,
                 )
+            if unassigned_users:
+                task._task_message_unsubscribe_notify(unassigned_users)
+
+    def _task_message_unsubscribe_notify(self, unassigned_users):
+        self.ensure_one()
+        unassigned_user_names = ', '.join(unassigned_users.mapped('name'))
+        subtype_id_unassigned = self.env['ir.model.data']._xmlid_to_res_id('project.mt_task_unassigned', raise_if_not_found=False)
+        followers_with_subtype = self.message_follower_ids.filtered(lambda follower: subtype_id_unassigned in follower.subtype_ids.ids)
+        if not followers_with_subtype:
+            return
+        values = {
+            'object': self,
+            'model_description': self.env['ir.model']._get(self._name).display_name,
+            'access_link': self._notify_get_action_link('view'),
+            'unassigned_user_names': unassigned_user_names,
+        }
+        unassignation_msg = self.env['ir.qweb']._render('project.project_message_user_unassigned', values, minimal_qcontext=True)
+        unassignation_msg = self.env['mail.render.mixin']._replace_local_links(unassignation_msg)
+        self.message_notify(
+            subject=_("%(unassignee_name)s has been unassigned from %(task_name)s", unassignee_name=unassigned_user_names, task_name=self.display_name),
```

Comment:
Not grammatically correct if there are multiple users. It would also be nice to rephrase this in a gender neutral way. Maybe "Users have been unassigned from ..."?